lib/babe/babe.go

```diff
@@ -190,38 +190,39 @@ func (b *Session) invokeBlockAuthoring() {
 }
 
 func (b *Session) handleSlot(slotNum uint64) {
-	parentHeader := b.blockState.LatestHeader()
+	parentHeader, err := b.blockState.BestBlockHeader()
+	if err != nil {
+		log.Error("BABE block authoring", "error", "parent header is nil")
+		return
```

Comment:
if we just return silently, how can the caller method know there was a failure and act on it ?

Reply:
the caller method is `invokeBlockAuthoring` which is called in a goroutine inside `Start()`. the caller method only handles errors by logging them, it doesn't return anything to `Start()`. 

also, it's fine if BABE can't produce a block for one slot.

Reply:
oh right, make sense